portal/src/main/webapp/js/src/pancancer_study_summary/pancancer_study_summary.js

```diff
@@ -790,12 +790,20 @@ function DataManagerPresenter(dmInitCallBack)
 			
 			//do the next call:
 			console.log(new Date() + ": CALL to get sample clinical atttributes (cancer types)");
-			return window.QuerySession.getSampleClinicalData(["CANCER_TYPE","CANCER_TYPE_DETAILED"]);
+			//this and next call below are in parallel, so the last one should call the dmInitCallBack:
+			nrResponsesReceived++;
+			if (nrResponsesReceived == 2) {
+				dmInitCallBack(self);
+			}
```

Comment:
I think I wasn't clear enough - my solution accounts for this problem. You resolve the promises after your individual callbacks have been called. So it really is functionally equivalent to what you have.

A function that does this might look like:
```
var promiseHandler = function(promiseA, promiseB, callbackA, callbackB, finally) {
    var callbackA_done = new $.Deferred();
    var callbackB_done = new $.Deferred();
    promiseA.then(function(response) {
        callbackA(response);
        callbackA_done.resolve();
    });
    promiseB.then(function(response) {
        callbackB(response);
        callbackB_done.resolve();
    });
    $.when(callbackA_done, callbackB_done).then(finally);
}
```